src/containers/LoginRegister/index.jsx

```diff
@@ -0,0 +1,201 @@
+      setLoading(true);
+      await auth.sendPasswordResetEmail(email);
+      setLoading(false);
+      setMessage(
+        <Alert
+          className="authAlert"
+          showIcon
+          message={
+            <div>
+              {t("loginRegister.anEmailSent")}
+              <b>{email}</b>
+              {t("loginRegister.anEmailSentContain")}
+            </div>
+          }
+          type="success"
+        />
+      );
+      setActive("login");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  const loginWithFacebook = async () => {
+    resetErrorAndMessage();
+    try {
+      setLoading(true);
+      const provider = new firebase.auth.FacebookAuthProvider();
+      provider.setCustomParameters({
+        display: "popup",
+      });
+      setLoading(false);
+      history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  const loginWithGoogle = async () => {
+    resetErrorAndMessage();
+    try {
+      setLoading(true);
+      const provider = new firebase.auth.GoogleAuthProvider();
+      provider.addScope("https://www.googleapis.com/auth/contacts.readonly");
+      setLoading(false);
+      history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  const register = async (fullName, email, age, password) => {
+    resetErrorAndMessage();
+    try {
+      setLoading(true);
+      const registeredUser = await auth.createUserWithEmailAndPassword(
+        email,
+        password
+      );
+      await firestore.collection("users").doc(registeredUser.user.uid).set({
+        uid: registeredUser.user.uid,
+        fullName,
+        age,
+      });
+      await auth.signInWithEmailAndPassword(email, password);
+      setLoading(false);
+      history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  let component;
+  switch (active) {
+    case "login":
+      component = (
+        <Login
+          error={error}
+          message={message}
+          onSubmit={({ email, password }) => {
+            login(email, password);
+          }}
+          handleRegister={() => {
+            resetErrorAndMessage();
+            setActive("register");
+          }}
+          handlePasswordReset={() => {
+            resetErrorAndMessage();
+            setActive("reset");
+          }}
+          handleFacebookAuth={() => {
+            resetErrorAndMessage();
+            loginWithFacebook();
+          }}
+          handleGoogleAuth={() => {
+            resetErrorAndMessage();
+            loginWithGoogle();
+          }}
+        />
+      );
+      break;
+    case "register":
+      component = (
+        <Register
+          error={error}
+          message={message}
+          onSubmit={({ fullName, email, age, password }) => {
+            register(fullName, email, age, password);
+          }}
+          handleLogin={() => {
```

Comment:
There is also another function named as handleLogin between 172-185.

Reply:
Thanks for pointing out. Repetitive functions are eliminated.